IABTechLab/uid2-self-serve-portal
# API user role checks

- Replace any `isApprover` API permission checks with `isUid2Support`.
- Protect certain endpoints from users without Admin or UID2 Support roles.
- Move shared auth checks to `userRoleMiddleware.ts`

## Manual testing

1. With the UID2 Support role, you should be able to do anything in the portal, despite not being in the Approvers table (though the Participants Awaiting Approval still relies on the Approvers table)
2. Remove yourself from the UID2 Support role, giving yourself only Admin
3. Modify the conditions in the front end code so that the team member management, manage participant, and audit trail screens are all reachable by applying [show-all-restricted-UI-actions.patch](https://github.com/user-attachments/files/17551746/show-all-restricted-UI-actions.patch). Alternatively call the endpoints via Postman
4. Admin should still be able to do everything expect Manage Participants
5. Remove yourself from Admin, giving yourself only Operations
6. You should no longer be able to manage (add/edit/remove/reinvite) team members or view audit trails

### Example: Removing a team member when the user only has `Operations`

https://github.com/user-attachments/assets/fd1e3f97-5788-433d-9058-39dc6bea84c8

Note: the front-end code has been modified so that the action buttons still appear.
